sdk/openai/openai/src/api/util.ts

```diff
@@ -87,3 +91,24 @@ function toSnakeCase<P extends string>(str: P): SnakeCase<P> {
     .replace(/([A-Z])/g, (group) => `_${group.toLowerCase()}`)
     .replace(/^_/, "") as SnakeCase<P>;
 }
+
+function statusCodeToNumber(statusCode: string): number | undefined {
+  const status = Number.parseInt(statusCode);
+
+  return Number.isNaN(status) ? undefined : status;
+}
+
+export function createOpenAIError(result: PathUncheckedResponse): OpenAIError {
+  const message = result.body.error?.message ?? "An error has occurred";
+  return new OpenAIError(message, {
+    code: result.body.error?.code,
+    param: result.body.error?.param,
+    type: result.body.error?.type,
+    statusCode: statusCodeToNumber(result.status),
+    response: {
+      headers: createHttpHeaders(result.headers),
+      request: result.request,
+      status: statusCodeToNumber(result.status) ?? -1,
+    },
+  });
```

Comment:
I think this can be simplified as follows:
```suggestion
  const err = result.body.error ?? result.body;
  if (!err) {
    throw new Error("An error response has been received but can't be parsed");
  }
  const statusCode = statusCodeToNumber(result.status);
  return new OpenAIError(err.message, {
    code: err.code,
    param: err.param,
    type: err.type,
    statusCode,
    response: {
      headers: createHttpHeaders(result.headers),
      request: result.request,
      status: statusCode ?? -1,
    },
  });
```